articles/object-arrays-destructuring/index.md

```diff
@@ -0,0 +1,233 @@
+
+### 3. Array destructuring
+
+Destructuring is the extracting object properties or array items into independent variables. Destructuring of arrays can be done like this:
+
+```javascript
+var fruits = [
+	"Bananas",
+	"Lemons",
+	"Mangoes"
+];
+
+var [fruit1, fruit2, fruit3] = fruits;
+```
+All the fruit items in `fruits` array above are mapped onto each variable defined in the last line. For example, `"Bananas"` is mapped onto `fruit1`. You can also skip values you are not interested in like this:
+
+```javascript
+var fruits = [
+	"Bananas",
+	"Lemons",
+	"Mangoes"
+];
+var [fruit1, , fruit3] = fruits;// skips lemons
+```
+
+Destructuring can also be used to define variables in bulk.
+
+```javascript
+const [fruit1, fruit2, fruit3] = ["Bananas", "Lemons", "Mangoes"];
+```
+
+Variable names can be declared separately from their values to achieve readability.
+
+```javascript
+let fruit1, fruit2, fruit3;
+
+[fruit1, fruit2, fruit3] = ["Bananas", "Lemons", "Mangoes"];
+```
+
+### 4. Object destructuring
+
+Destructuring in JavaScript objects can be done as follows:
+
+```javascript
+var person = {
+	name : "Geoffrey",
+	role : "Student"
+};
+
+const {name, role} = person;
+```
+The above method only works when the variable names are the same as the object property names. To assign object properties to new variable names, declare them like this.
+
+```javascript
+const {name:naming, role:roling} = person;
+```
+
+### 5. Swapping variables using destructuring
+
+Destructuring is a common way of doing complex or repetitive work more easily. Besides defining variables in bulk, it can also be used to swap variables.
+
+The commonly used method of swapping variables in JavaScript is by introducing a temporary variable. The temporary variable holds the value of a variable as it gets swapped.
+
+```javascript
+let a = 1;
+let b = 2;
+let temp;
+
+temp = a;
+a = b;
+b = temp;
+```
+
+Destructuring can be used to do the same thing with fewer lines of code.
+
+```javascript
+let a = 1;
+let b = 2;
+
+[a, b] = [b, a];
+```
+
+### Conclusion
+
+Objects are a data-type that has properties or preferences associated with it. If you have worked with JSON data, you have already worked with objects. Arrays are a data-type used to store related items or creating ordered lists. Since objects and arrays build a fraction of a web developer's code, understanding the basic concepts around these data types is important. 
```

Comment:
Try rewording the last sentence for readability. Maybe say "make up" instead of "build".